packages/blade/src/components/Input/PasswordField/PasswordField.tsx

```diff
@@ -0,0 +1,151 @@
+
+type Maybe<Type> = Type | undefined;
+
+type PasswordFieldExtraProps = {
+  /**
+   * Shows a reveal button to toggle password visibility
+   *
+   * @default true
+   */
+  showRevealButton?: boolean;
+
+  /**
+   * Displays asterisk (`*`) when `isRequired` is enabled
+   *
+   * @default none
+   */
+  necessityIndicator?: Exclude<BaseInputProps['necessityIndicator'], 'optional'>;
+
+  /**
+   * Determines what autoComplete suggestion type to show. Defaults to using platform heuristics.
+   *
+   * It's not recommended to turn this off in favor of safe password practices.
+   * Providing `password` or `newPassword` is more informative to the platform for browser autofill or password managers.
+   *
+   * **Note**: Using `newPassword` on iOS has some [known issue](https://github.com/facebook/react-native/issues/21911) on React Native
+   *
+   * Internally it'll render platform specific attributes:
+   *
+   * - web: `autocomplete`
+   * - iOS: `textContentType`
+   * - android: `autoComplete`
+   *
+   */
+  autoCompleteSuggestionType?: Extract<
+    BaseInputProps['autoCompleteSuggestionType'],
+    'none' | 'password' | 'newPassword'
+  >;
+};
+
+type PasswordFieldProps = Pick<
+  BaseInputProps,
+  | 'id'
+  | 'label'
+  | 'labelPosition'
+  | 'maxCharacters'
+  | 'validationState'
+  | 'errorText'
+  | 'successText'
+  | 'helpText'
+  | 'isDisabled'
+  | 'defaultValue'
+  | 'placeholder'
+  | 'isRequired'
+  | 'value'
+  | 'onChange'
+  | 'onBlur'
+  | 'onFocus'
+  | 'name'
+  | 'autoFocus'
+  | 'keyboardReturnKeyType'
+  | 'autoCompleteSuggestionType'
+> &
+  PasswordFieldExtraProps;
+
+const PasswordField = ({
+  id,
+  label,
+  labelPosition = 'top',
+  showRevealButton = true,
+  maxCharacters,
+  validationState,
+  errorText,
+  successText,
+  helpText,
+  isDisabled = false,
+  defaultValue,
+  placeholder,
+  isRequired = false,
+  necessityIndicator = 'none',
+  value,
+  onChange,
+  onFocus,
+  onBlur,
+  name,
+  autoFocus = false,
+  keyboardReturnKeyType = 'done',
+  autoCompleteSuggestionType,
+}: PasswordFieldProps): ReactElement => {
+  const [isRevealed, setIsRevealed] = useState(false);
+
+  const toggleIsRevealed = (): void => setIsRevealed((revealed) => !revealed);
+  const accessibilityLabel = isRevealed ? 'Hide password' : 'Show password';
+  const type = isRevealed ? 'text' : 'password';
+
+  const revealButtonIcon = isRevealed ? EyeOffIcon : EyeIcon;
+  const revealButton = showRevealButton ? (
+    <IconButton
+      size="medium"
+      icon={revealButtonIcon}
+      onClick={toggleIsRevealed}
+      accessibilityLabel={accessibilityLabel}
+    />
+  ) : null;
+
+  const trailingFooterSlot = (value: Maybe<string>): Maybe<ReactNode> =>
+    maxCharacters ? (
+      <Box marginTop="spacing.2" marginRight="spacing.1">
+        <CharacterCounter currentCount={value?.length ?? 0} maxCount={maxCharacters} />
+      </Box>
+    ) : null;
+
+  return (
+    <BaseInput
+      id={id}
```

Comment:
I think this ID is not meant to be passed by users 

ref: 
https://razorpay.slack.com/archives/G01B3LQ9H0W/p1663568715561059